src/MainUpdate.elm

```diff
@@ -510,3 +511,52 @@ update msg model =
 
             GetDateAndThenAddDate id ->
                 model ! [ Task.perform (AddNewDate id) Date.now ]
+
+            DeleteTrack trackId ->
+                let
+                    newTracks =
+                        List.filter (\t -> t.id /= trackId) model.tracks
+
+                    apiUpdate =
+                        { datesWithSessions = model.datesWithSessions
+                        , tracks = newTracks
+                        , columns = model.columns
+                        }
+                in
+                    ( { model | tracks = newTracks }
+                    , Api.postModelToDb apiUpdate model.eventId
+                    )
+
+            AddNewTrack ->
+                ( { model
+                    | pickedTracks =
+                        appendNewElementToList model.pickedTracks (Track 0 "" "")
+                  }
+                , Cmd.none
+                )
+
+            UpdatePickedTrack trackId trackInputType newPickedTrackInput ->
+                let
+                    pickedTrack =
+                        model.pickedTracks
+                            |> List.filter (\t -> t.id == trackId)
+                            |> List.head
+                            |> Maybe.withDefault (Track 0 "" "")
+
+                    newPickedTrack =
+                        if trackInputType == "name" then
+                            { pickedTrack
+                                | name = newPickedTrackInput
+                            }
+                        else if trackInputType == "description" then
+                            { pickedTrack
+                                | description = newPickedTrackInput
+                            }
+                        else
+                            pickedTrack
+
+                    pickedTracksWithoutUpdatedTrack =
+                        model.pickedTracks
+                            |> List.filter (\s -> s.id /= trackId)
+                in
+                    ( { model | pickedTracks = List.append pickedTracksWithoutUpdatedTrack [ newPickedTrack ] }, Cmd.none )
```

Comment:
Doesn't need to be changed but it's good to get into the habit of using cons `::` over append if possible as it runs in constant time